docs/references/commands.txt

```diff
@@ -0,0 +1,326 @@
+..
+   THIS DOCUMENTATION WAS AUTOMATICALLY GENERATED
+   Do not edit this document directly. Instead, edit the commands in the cli
+   source code, which can be found on the main branch of the git repo at
+   https://github.com/canonical/ubuntu-pro-client. The cli source code is
+   nested in the uaclient/cli folder. Description for the commands is found
+   in the uaclient/messages folder.
+
+Available commands
+==================
+
+The currently available commands are:
+
+- `pro api`_
+- `pro attach`_
+- `pro auto-attach`_
+- `pro collect-logs`_
+- `pro config`_
+- `pro detach`_
+- `pro disable`_
+- `pro enable`_
+- `pro fix`_
+- `pro help`_
+- `pro refresh`_
+- `pro security-status`_
+- `pro status`_
+- `pro system`_
+
+
+pro api
+=======
+
+**Usage:**
+
+``pro api [-h] [--show-progress] [--args [OPTIONS ...]] [--data DATA] endpoint``
+
+**Description:**
+
+Calls the Client API endpoints.
+
+For a list of all of the supported endpoints and their structure,
+please refer to the Pro Client API reference guide:
+
+https://canonical-ubuntu-pro-client.readthedocs-hosted.com/en/latest/references/api/
+
+pro attach
+==========
+
+**Usage:**
+
+``pro attach [-h] [--no-auto-enable] [--attach-config ATTACH_CONFIG] [--format {cli,json}] [token]``
+
+**Description:**
+
+Attach this machine to an Ubuntu Pro subscription with a token obtained from:
+https://ubuntu.com/pro/dashboard
+
+When running this command without a token, it will generate a short code
+and prompt you to attach the machine to your Ubuntu Pro account using
+a web browser.
+
+The --attach-config option can be used to provide a file with the token
+and optionally, a list of services to enable after attaching. To know more,
+visit:
+https://canonical-ubuntu-pro-client.readthedocs-hosted.com/en/latest/howtoguides/how_to_attach_with_config_file/
+
+The exit code will be:
+
+    * 0: on successful attach
+    * 1: in case of any error while trying to attach
+    * 2: if the machine is already attached
+
+pro auto-attach
+===============
+
+**Usage:**
+
+``pro auto-attach [-h]``
+
+**Description:**
+
+Automatically attach on an Ubuntu Pro cloud instance.
+
+pro collect-logs
+================
+
+**Usage:**
+
+``pro collect-logs [-h] [-o OUTPUT]``
+
+**Description:**
+
+Collect logs and relevant system information into a tarball.
+This information can be later used for triaging/debugging issues.
+
+pro config
+==========
+
+**Usage:**
+
+``pro config [-h] {show,set,unset} ...``
+
+**Description:**
+
+Manage Ubuntu Pro Client configuration on this machine
+
+pro config show
+---------------
+
+**Usage:**
+
+``pro config show [-h] [key]``
+
+**Description:**
+
+Show customizable configuration settings
+
+pro config set
+--------------
+
+**Usage:**
+
+``pro config set [-h] key_value_pair``
+
+**Description:**
+
+Set and apply Ubuntu Pro configuration settings
```

Comment:
```suggestion
Set and apply Ubuntu Pro configuration settings.
```

Reply:
Is there a list anywhere of what keys can be set, and what values are accepted?

Reply:
yes, but right now we don't have it in the base descriptions used here. This may fall part of Grant's suggestion below on further explaining the flags and options.